scanpy/external/pl.py

```diff
@@ -135,3 +137,103 @@ def harmony_timeseries(
         return fig
     elif not show:
         return axes
+
+
+def sam(
+    adata: AnnData,
+    projection: Union[str, np.ndarray] = 'X_umap',
+    c: Optional[Union[str, np.ndarray]] = None,
+    cmap: str = 'rainbow',
```

Comment:
Please never use rainbow: https://eagereyes.org/basics/rainbow-color-map

```suggestion
    cmap: str = 'viridis',
```